src/attention/utils/helpers.ts

```diff
@@ -163,20 +163,31 @@ export async function useRecompute(state: AttentionState) {
     middleware: [
       offset({ mainAxis: state?.distance ?? 8, crossAxis: state?.skidding ?? 0}), // offers flexibility over how to place the attentionEl towards its targetEl both on the x and y axis (horizontally and vertically).
       state?.flip && flip({ //when flip is set to true it will move the attentionEl's placement to its opposite side or to the preferred placements if fallbackPlacements has a value
-        fallbackAxisSideDirection: 'start', // the preferred placement axis fit when flip is set to true and fallbackPlacements does not have a value. 'start' represents 'top' or 'left'.
+        crossAxis: state?.crossAxis, //checks overflow to trigger a flip. When disabled, it will ignore overflow
         fallbackPlacements: state?.fallbackPlacements,
       }),
-      !state?.flip && autoPlacement(), // when flip is set to false, it will instead call autoPlacement() that will choose the placement that has the most space available automatically. You can only use either flip() or autoPlacement(), and not combined. 
-      shift({ padding: 16}),
       !state?.noArrow && state?.arrowEl && arrow({ element: state?.arrowEl }),
+      hide({ //will hide the attentionEl when it appears detached from the targetEl. Can be called multiple times in the middleware-array if you want to use several strategies
+        strategy: 'escaped', //default strategy is 'referenceHidden'
+      }),
+      hide(), // we call hide() again here to also trigger the 'referenceHidden' strategy. 
     ],
   }).then(({ x, y, middlewareData, placement }) => {
     state.actualDirection = placement
     
     Object.assign(attentionEl?.style, {
       left: `${x}px`,
       top: `${y}px`,
-    })
+    });
+
+    if (middlewareData?.hide && !state?.isCallout) {      
+      const { escaped, referenceHidden } = middlewareData?.hide
+      Object.assign(attentionEl?.style, {
+        visibility: referenceHidden ? 'hidden' : '',
```

Comment:
Don't we need to set `'visible'` if not `'hidden'`?

Reply:
The attention component has a `display: block` that I think makes it visible. If we want to set it to `visible` if not `hidden`, then I need to revert removing `!state?.isTooltip` from the if-statement, since then the tooltip will have a `visibility: visible` on it, even when you are not hovering on the `targetEl`, meaning that the tooltip will be visible regardless of when hovering or not.

Reply:
I thought `visibility: visible` would only be applied if `isShowing` was also `true`. I also thought the hiding logic also be triggered only if the attention was originally showing 😅

Reply:
When rendering the page where the attention component is and the `isShowing` is first set to `false`, then we will arrive to line 150 `if(!state?.isShowing) return` so then it will not set `visibility:hidden`. However, once `isShowing` is set to `true`, which makes the attention component visible, it will get `visibility:visible`, if we have it like this:
```
if (middlewareData?.hide && !state?.isCallout) {      
      const { escaped, referenceHidden } = middlewareData?.hide
      Object.assign(attentionEl?.style, {
        visibility: referenceHidden ? 'hidden' : 'visible',
        opacity: escaped ? '0.5' : '',
      })
    }
```
Therefore, when `targetEl` is still in viewport, the `attentionEl` will have `visibility: visible`. It will not get `visibility: hidden` until `targetEl` is no longer in viewport, unless it is a `callout`. This means, that if you hover on the `targetEl` to trigger the `tooltip`, it will show up on hover, but it will not hide the `tooltip` when unhovering the `targetEl`, unless `targetEl` is no longer in viewport. :)

Reply:
Makes sense! Thanks for explaining :D